File: 프로그래머스/1/42862. 체육복/체육복.java
import java.util.*;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        Arrays.sort(lost);
        Arrays.sort(reserve);
        List<Integer> losts = new ArrayList<>();
        List<Integer> reserves = new ArrayList<>();
        List<Integer> needToRemove = new ArrayList<>();
        
        for (int l: lost) losts.add(l);
        for (int r: reserve) reserves.add(r);
        
        for (int l : losts) {
            if (reserves.contains(l)) {
                needToRemove.add(l);
            }
        }
        
        for (int t: needToRemove) {
            losts.remove(Integer.valueOf(t));
            reserves.remove(Integer.valueOf(t));
        }
        
        
        int answer = n;
        
        while (losts.size() > 0 && reserves.size() > 0) {
            if (losts.get(0) - 1 > reserves.get(0)) {
                reserves.remove(0);
            } else if (losts.get(0) + 1 < reserves.get(0)) {
                answer -= 1;
                losts.remove(0);
            } else if (losts.get(0) - 1 == reserves.get(0) || losts.get(0) + 1 == reserves.get(0)) {
                reserves.remove(0);
                losts.remove(0);
            }
        }
        
        answer -= losts.size();
        return answer;
    }
}

File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    answer = n
    lost.sort()
    reserve.sort()
    
    need_to_remove = []
    for l in lost:
        if l in reserve:
            need_to_remove.append(l)
            
    for n in need_to_remove:
        lost.remove(n)
        reserve.remove(n)
    
    while lost and reserve:
        if lost[0] - 1 > reserve[0]:
            reserve.pop(0)
        elif lost[0] + 1 < reserve[0]:
            answer -= 1
            lost.pop(0)
        elif lost[0] + 1 == reserve[0] or lost[0] - 1 == reserve[0]:
            lost.pop(0)
            reserve.pop(0)
    
    answer -= len(lost)
            
    return answer